code/game/objects/items/devices/painter/painter.dm

```diff
@@ -0,0 +1,70 @@
+/// Base painter object (Should not be spawned)
+/obj/item/painter
+	name = "modular painter"
+	icon = 'icons/obj/painting.dmi'
+	usesound = 'sound/effects/spray2.ogg'
+	flags = CONDUCT | NOBLUDGEON
+	w_class = WEIGHT_CLASS_SMALL
+	slot_flags = SLOT_BELT
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 2000)
+
+	var/static/list/painter_types = list(
+		"Floor Painter" = /obj/item/painter/floor,
+		"Pipe Painter" = /obj/item/painter/pipe,
+		"Window Painter" = /obj/item/painter/pipe/window,
+		"Airlock Painter" = /obj/item/painter/airlock)
+
+/obj/item/painter/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Ctrl+click it in your hand to change the type!</span>"
+
+/obj/item/painter/CtrlClick(mob/user)
+	. = ..()
+	if(!ishuman(user)) // Only want human players doing this
+		return
+	if(loc != user) // Not being held
+		return
+
+	playsound(src, 'sound/effects/pop.ogg', 50, TRUE)
+	var/choice = show_radial_menu(user, src, painter_types, require_near = TRUE)
+	if(!choice)
+		return
+	var/choice_path = painter_types[choice]
+	if(istype(src, choice_path))
+		return
+
+	playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+	do_sparks(1, FALSE, src)
+	var/obj/item/painter/P = new choice_path(get_turf(user))
+	if(P)
+		qdel(src) // Swap it out for the new one
```

Comment:
I'm skeptical of new'ing and qdel'ing the items each time you want to switch. I know other items do it to like the jaws of life, but it feels hacky. 

I feel like a better way to do it would be with a `mode` var or similar. For example, given the target is an airlock, in `afterattack` you check it's `mode`. If its in airlock mode, then paint the airlock. Then you can unify everything under one item, sorta like the RCD.

Reply:
The reason I didn't do something like this is because of all of variables that the different painter types need. Having them defined on the main object or inside `afterattack()` would get messy really quickly, whereas having them as subtypes overriding a proc is much cleaner.
The new/qdel thing is definitely an issue though. I'd considered having some sort of variable to store any unused painters once they've been spawned to avoid having to delete them, but I wasn't sure how that would work out.

Reply:
Yeah, I mean I suppose you could just store all 4 painter types inside of the modular painter, and have 4 variable as references to track them. You check the type of object the painter hit, and then route it to the respective painter which can handle it.

Reply:
I'm having a lot of trouble trying to figure out how to do this properly, so I'll commit the other two separately.